R/messages.R

```diff
@@ -0,0 +1,72 @@
+#### Code for interacting with the hector messaging system
+
+
+default_fetchvars <- c(ATMOSPHERIC_CO2, RF_TOTAL, RF_CO2, GLOBAL_TEMP)
```

Comment:
`format.hcore` is the specialization of the `format` generic to the `hcore` class.  It _has_ to be named that way, per the S language specification (of which R is an implementation).

`default_fetchvars` is an object that is _not_ a specialization of a generic, so it doesn't use a `.` as a word separator so as to avoid giving the impression that it might be.  Base R doesn't use this convention, and you see names like `read.csv` all over the place, but tidyverse conventions specify that the `.` should be reserved for specializing generics, and I agree with them.

`isactive` is kind of a special case.  The convention of using `isfoo` to name a predicate testing the "foo" property goes back at least to the 1990s (*).  Prior to that you probably would have used `foop`, but these days only programmers of advanced age and thinning hairline like me are familiar with that usage.  You see this some in base R (`isS3method`, `isGeneric`, etc.), but it's inconsistent.  The `is` function is a generic for testing class membership, so `is.foo` should be reserved for specializations of that method.  Tidyverse style would suggest `is_foo`.  Personally, I think that's needless extra typing, and simply running the "is" together is more readable to me, but then I've been reading code written this way for a very long time.

I'll change `isactive` to `is_active` if you really feel strongly about it.  The other two are pretty much dictated by the structure of the language.

(*) If, however, the property being tested is a verb, you normally drop the "is", as in `inherits`.

Reply:
Ah, right about `format.hcore`, missed that. I prefer `is_active` but ymmv so either.